From c15c45c54bb2fd03b41416c6628efcb095eec4b6 Mon Sep 17 00:00:00 2001
From: Xaver Hugl <xaver.hugl@kde.org>
Date: Mon, 17 Feb 2025 16:25:09 +0100
Subject: [PATCH 1/4] increase required wayland-protocols version to 1.41

This version contains the color management protocol.

This commit also adjusts the build-wayland script to mention
that the DEBIAN_BASE_TAG also has to be bumped.

Signed-off-by: Xaver Hugl <xaver.hugl@kde.org>
Reviewed-by: Eric Engestrom <eric@igalia.com>
Reviewed-by: Sebastian Wick <sebastian.wick@redhat.com>
Acked-by: Daniel Stone <daniels@collabora.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/32038>
---
 .gitlab-ci/container/build-wayland.sh |  4 +++-
 .gitlab-ci/image-tags.yml             | 16 ++++++++--------
 meson.build                           |  2 +-
 subprojects/wayland-protocols.wrap    |  6 +++---
 4 files changed, 15 insertions(+), 13 deletions(-)

diff --git a/.gitlab-ci/container/build-wayland.sh b/.gitlab-ci/container/build-wayland.sh
index d6117254e87..551edcfe7db 100644
--- a/.gitlab-ci/container/build-wayland.sh
+++ b/.gitlab-ci/container/build-wayland.sh
@@ -7,6 +7,8 @@ uncollapsed_section_start wayland "Building Wayland"
 
 # When changing this file, you need to bump the following
 # .gitlab-ci/image-tags.yml tags:
+# ALPINE_X86_64_BUILD_TAG
+# DEBIAN_BASE_TAG
 # DEBIAN_BUILD_TAG
 # DEBIAN_TEST_ANDROID_TAG
 # DEBIAN_TEST_GL_TAG
@@ -15,7 +17,7 @@ uncollapsed_section_start wayland "Building Wayland"
 # KERNEL_ROOTFS_TAG
 
 export LIBWAYLAND_VERSION="1.21.0"
-export WAYLAND_PROTOCOLS_VERSION="1.38"
+export WAYLAND_PROTOCOLS_VERSION="1.41"
 
 git clone https://gitlab.freedesktop.org/wayland/wayland
 cd wayland
diff --git a/.gitlab-ci/image-tags.yml b/.gitlab-ci/image-tags.yml
index 89909ed7912..b8423ef3d4f 100644
--- a/.gitlab-ci/image-tags.yml
+++ b/.gitlab-ci/image-tags.yml
@@ -13,10 +13,10 @@
 
 variables:
    DEBIAN_X86_64_BUILD_BASE_IMAGE: "debian/x86_64_build-base"
-   DEBIAN_BASE_TAG: "20250130-vvless"
+   DEBIAN_BASE_TAG: "20250223-way-prot"
 
    DEBIAN_X86_64_BUILD_IMAGE_PATH: "debian/x86_64_build"
-   DEBIAN_BUILD_TAG: "20250128-llvm19-up"
+   DEBIAN_BUILD_TAG: "20250223-way-prot"
 
    DEBIAN_X86_64_TEST_BASE_IMAGE: "debian/x86_64_test-base"
    DEBIAN_ARM64_TEST_BASE_IMAGE: "debian/arm64_test-base"
@@ -27,17 +27,17 @@ variables:
    DEBIAN_ARM64_TEST_IMAGE_VK_PATH: "debian/arm64_test-vk"
    DEBIAN_X86_64_TEST_ANDROID_IMAGE_PATH: "debian/x86_64_test-android"
 
-   DEBIAN_TEST_ANDROID_TAG: "20250130-vvless"
-   DEBIAN_TEST_GL_TAG: "20250327-piglit-250"
-   DEBIAN_TEST_VK_TAG: "20250327-piglit-250"
-   KERNEL_ROOTFS_TAG: "20250327-trace-250"
+   DEBIAN_TEST_ANDROID_TAG: "20250223-way-prot"
+   DEBIAN_TEST_GL_TAG: "20250223-way-prot"
+   DEBIAN_TEST_VK_TAG: "20250223-way-prot"
+   KERNEL_ROOTFS_TAG: "20250223-way-prot"
 
    DEBIAN_PYUTILS_IMAGE: "debian/x86_64_pyutils"
    DEBIAN_PYUTILS_TAG: "20250129-lavacli"
 
-   ALPINE_X86_64_BUILD_TAG: "20250128-llversionm"
+   ALPINE_X86_64_BUILD_TAG: "20250223-way-prot"
    ALPINE_X86_64_LAVA_SSH_TAG: "20250124-spirv-tools"
-   FEDORA_X86_64_BUILD_TAG: "20250128-llversionm"
+   FEDORA_X86_64_BUILD_TAG: "20250223-way-prot"
 
    KERNEL_TAG: "v6.13-rc4-mesa-5e77"
    KERNEL_REPO: "gfx-ci/linux"
diff --git a/meson.build b/meson.build
index 095aec5d9ea..ddde02c1755 100644
--- a/meson.build
+++ b/meson.build
@@ -2058,7 +2058,7 @@ if with_platform_wayland
   else
     wl_scanner_arg = 'code'
   endif
-  dep_wl_protocols = dependency('wayland-protocols', version : '>= 1.38', default_options: [ 'tests=false' ])
+  dep_wl_protocols = dependency('wayland-protocols', version : '>= 1.41', default_options: [ 'tests=false' ])
   dep_wayland_client = dependency('wayland-client', version : '>=1.18')
   dep_wayland_server = dependency('wayland-server', version : '>=1.18')
   if with_egl
diff --git a/subprojects/wayland-protocols.wrap b/subprojects/wayland-protocols.wrap
index 61f306bdffd..c82fece01e3 100644
--- a/subprojects/wayland-protocols.wrap
+++ b/subprojects/wayland-protocols.wrap
@@ -1,7 +1,7 @@
 [wrap-file]
-directory = wayland-protocols-1.38
-source_url = https://gitlab.freedesktop.org/wayland/wayland-protocols/-/releases/1.38/downloads/wayland-protocols-1.38.tar.xz
-source_filename = wayland-protocols-1.38.tar.xz
+directory = wayland-protocols-1.41
+source_url = https://gitlab.freedesktop.org/wayland/wayland-protocols/-/releases/1.41/downloads/wayland-protocols-1.41.tar.xz
+source_filename = wayland-protocols-1.41.tar.xz
 source_hash = ff17292c05159d2b20ce6cacfe42d7e31a28198fa1429a769b03af7c38581dbe
 
 [provide]
-- 
2.47.0

